### ISSUE TITLE
[msal-node-extensions] cannot import MSAL Node Extensions into an ES Module on Node

### ISSUE THREAD

**Issue by jetwiwo** [AzureAD/microsoft-authentication-library-for-js#6573]
### Core Library

MSAL Node (@azure/msal-node)

### Core Library Version

2.2.0

### Wrapper Library

MSAL Node Extensions (@azure/msal-node-extensions)

### Wrapper Library Version

1.0.4

### Public or Confidential Client?

Public

### Description

Cannot import `@azure/msal-node-extensions` from an ES Module in Node (I've tried Node 18, 20, and 21).

It appears that the MJS version of `KeychainPersistence` is importing from `keytar` which is a CommonJS module and can't be imported by ESM.

### Error Message

```
$ node minimal.mjs
file:///D:/experiments/msal-with-cache-esm/node_modules/@azure/msal-node-extensions/dist/persistence/KeychainPersistence.mjs:3
import { setPassword, getPassword, deletePassword } from 'keytar';
                                   ^^^^^^^^^^^^^^
SyntaxError: Named export 'deletePassword' not found. The requested module 'keytar' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'keytar';
const { setPassword, getPassword, deletePassword } = pkg;
```

### Msal Logs

N/A. MSAL does not start.

### MSAL Configuration

```javascript
N/A. No configuration required to reproduce this as I can't even import the library.
```

### Relevant Code Snippets

```javascript
// minimal.mjs
import { PersistenceCreator } from '@azure/msal-node-extensions';

console.log(typeof PersistenceCreator);
```

### Reproduction Steps

1. Create new node application (i.e. `npm init`)
2. Create `minimal.mjs` with the code snippet above
3. node `minimal.mjs`

### Expected Behavior

I should be able to use the MJS version of '@azure/msal-node-extensions' from an ES Module in Node.

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

None (Server)

### Source

External (Customer)

**Comment by rzyns** [comment]
I'm seeing a similar issue:

```
file:///Users/janusz/git/corechain/corechain/common/temp/node_modules/.pnpm/@azure+msal-node-extensions@1.0.4/node_modules/@azure/msal-node-extensions/dist/persistence/KeychainPersistence.mjs:3
import { setPassword, getPassword, deletePassword } from 'keytar';
                                   ^^^^^^^^^^^^^^
SyntaxError: The requested module 'keytar' does not provide an export named 'deletePassword'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:190:5)

Node.js v18.17.1
```

I get this same error regardless of whether I use a static or dynamic import, however if I monkey-patch [`LibSecretPersistence.ts`](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/9934e3b07118403898e65091035d26295e55f128/extensions/msal-node-extensions/src/persistence/LibSecretPersistence.ts#L6) (or, more accurately, `dist/persistence/LibSecretPersistence.mjs`) like so:

```typescript
// import { setPassword, getPassword, deletePassword } from 'keytar';
const keytar = await import("keytar");
const { setPassword, getPassword, deletePassword } = keytar.default;
```

then it works.

**Comment by tnorling** [comment]
Feel free to open a PR if you have a fix

**Comment by jetwiwo** [comment]
This issue is not yet resolved, but will likely be resolved by #6600 when it is released
